md2notion/NotionPyRenderer.py

```diff
@@ -14,6 +17,17 @@ def flatten(l):
         else:
             yield el
 
+def addHtmlImgTagExtension(notionPyRendererCls):
+    """A decorator that add the image tag extension to the argument list. The
+    decorator pattern allows us to chain multiple extensions. For example, we
+    can create a renderer with extension A, B, C by writing:
+        addAExtension(addBExtension(addCExtension(notionPyRendererCls)))
+    """
+    def newNotionPyRendererCls(*extraExtensions):
+        new_extension = [HTMLBlock, HTMLSpan]
+        return notionPyRendererCls(*chain(new_extension, extraExtensions))
+    return newNotionPyRendererCls
```

Comment:
How about using the decorator pattern?

Reply:
Hmm, any reason to do that versus creating a new class that inherits, and then just switch out the implementation in `upload.py` with the separate class? That's what my brain immediately defaulted to when I thought through the problem

```
class NotionPyRendererWithHTML(NotionPyRenderer):
    def __init__(*extraExtensions):
        super().__init__(*chain([HTMLBlock, HTMLSpan], extraExtensions)
```

Reply:
It could well be that in the future we add other extensions like the one in #20. Users can optionally turn them on or off. Then we have to provide class `NotionPyRenderer `, `NotionPyRendererWithHTML`, `NotionPyRendererWithEquation`, `NotionPyRendererWithHtmlAndEquation`. It becomes worse if there are three extensions.
Using the decorator pattern, we can write:
```python
renderer = NotionPyRenderer
if htmlEnabledByUser():
    renderer = addHtmlExtension(renderer)
if equationEnabledByUser():
    renderer = addEquationExtension(renderer)
```

Reply:
Ah yes, okay, then decorator pattern sounds good to me!